pynest/nest/ll_api.py

```diff
@@ -271,6 +272,54 @@ def check_stack(thing):
         raise ValueError("unable to decorate {0}".format(thing))
 
 
+class KernelAttribute:
+    """
+    Descriptor that dispatches attribute access to the nest kernel.
+    """
+    def __init__(self, typehint, doc, readonly=False, default=None, localonly=False):
+        self._readonly = readonly
+        self._localonly = localonly
+        self._default = default
+
+        self.__doc__ = (
+            doc
+            + ("." if default is None else f", defaults to ``{default}``.")
+            + ("\n\n" if readonly or localonly else "") + ", ".join(
+                c for c in (
+                    readonly and "**read only**",
+                    localonly and "**local only**",
+                )
+                if c
+            )
+            + f"\n\n:type: {typehint}"
+        )
```

Comment:
It took me quite some time to understand this code as it mixes quite a number of constructs (and this type of class-building is not a part of Python I have much experience with ...; I even learned something new about `and` in Python from this code). I tried to rewrite the code a little, including renaming `doc` to `description`, with the aim to make the code more expressive:

```suggestion
        self.__doc__ = "".join((
            description,
            "." if default is None else f", defaults to ``{default}``.",
            "\n\n" if readonly or localonly else "",
            ", ".join(c for c in (readonly and "**read only**",
                                          localonly and "**local only**") if c),
            f"\n\n:type: {typehint}"))
```

I don't like the format of the docstring we get here, with a "loose" "read only, local only" line, and the default somewhat hidden in the textual description, which may be long. Ideally, I would include default/ro/lo information in the title line, e.g.,

```
resolution [default: 0.1]

<description>

biologial_time [read-only]

<description>
```

If that is not possible in Sphinx, I'd suggest to put the line with default/ro/lo information directly below the title.

Reply:
Could we maybe postpone the formatting discussion to a follow-up PR? The whole PyNEST API page needs a serious overhaul anyway and having the information like it is now is already a huge improvement over what we had previously.

This PR already contains a lot of changes on the side and we want to release in two weeks.

Reply:
:+1:

Reply:
@jougs While I agree that we should not discuss formatting of the docstring here, I would still like you to consider my code suggestion above. I think the `join()` approach is clearer than the long sequence of string additions.